libjwt/jwt-gnutls.c

```diff
@@ -191,9 +191,17 @@ int jwt_sign_sha_pem(jwt_t *jwt, char **out, unsigned int *len, const char *str)
 	}
 
 	/* Sign data */
-        if (gnutls_privkey_sign_data2(privkey, alg, 0, &body_dat, &sig_dat)) {
-                ret = EINVAL;
-                goto sign_clean_privkey;
+        if (pk_alg == GNUTLS_PK_EC) {
+                /* ec signatues don't like gnutls_privkey_sign_data2 */
+                if (gnutls_privkey_sign_data(privkey, alg, 0, &body_dat, &sig_dat)) {
```

Comment:
Question: Does RSA do well with `gnutls_privkey_sign_data()`? If so, can we use that for both?

Reply:
`gnutls_privkey_sign_data` works well with RSA and ECDSA signatures but not with RSA-PSS.
`gnutls_privkey_sign_data2` works well with RSA and RSA-PSS signatures but not with ECDSA.

So we could use `gnutls_privkey_sign_data` with RSA as well if you prefer.